### PR TITLE
Add README for Golang image

### PR DESCRIPTION
**Pull request by alexandrevicenzi**
This is related to #993.

### CODE REVIEW

File: src/bci_build/package/golang/README.md.j2 [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# {{ image.pretty_name }} Container Image
+
+{% include 'badges.j2' %}
+
+## Description
+
+[Go](https://go.dev/) (a.k.a., Golang) is a programming language first developed at Google. It is a statically-typed language with syntax loosely derived from C, but with additional features such as garbage collection, type safety, some dynamic-typing capabilities, additional built-in types (e.g., variable-length arrays and key-value maps), and a large standard library.
```
Comment (dmpop) [inline src/bci_build/package/golang/README.md.j2:7]:
```suggestion
[Go](https://go.dev/) (a.k.a., Golang) is a statically-typed programming language, with syntax loosely derived from C. Go offers additional features such as garbage collection, type safety, certain dynamic-typing capabilities, additional built-in types (for example, variable-length arrays and key-value maps) as well as a large standard library.
```

File: src/bci_build/package/golang/README.md.j2 [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+## Usage
+
+The most straightforward way to use this image is to use a Go container as both the build and runtime environment.
+
+To compile and deploy your application, copy the sources, fetch dependencies (assuming go.mod for dependency management), and build the binary:
```
Comment (dmpop) [inline src/bci_build/package/golang/README.md.j2:13]:
```suggestion
To compile and deploy an application, copy the sources, fetch dependencies (assuming go.mod is used for dependency management), and build the binary:
```

File: src/bci_build/package/golang/README.md.j2 [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+COPY . .
+
+RUN go mod download && go mod verify
+RUN go build -v -o /usr/local/bin/app ./...
+
+CMD ["app"]
+```
+
+You can then build and run the container image:
```
Comment (dmpop) [inline src/bci_build/package/golang/README.md.j2:30]:
```suggestion
Build and run the container image:
```

File: src/bci_build/package/golang/README.md.j2 [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+```ShellSession
+$ podman build -t my-golang-app .
+$ podman run -it --rm my-golang-app
+```
+
+There may be occasions when it is inappropriate to run your application inside a container.
```
Comment (dmpop) [inline src/bci_build/package/golang/README.md.j2:37]:
```suggestion
There are situations when you don't want to run an application inside a container.
```

File: src/bci_build/package/golang/README.md.j2 [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+To compile, but not run your application inside a container instance, run the following command:
```
Comment (dmpop) [inline src/bci_build/package/golang/README.md.j2:39]:
```suggestion
To compile the application,  without running it inside a container instance, use the following command:
```

File: src/bci_build/package/golang/README.md.j2 [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+```ShellSession
+$ podman run --rm -v "$PWD":/app -w /app {{ image.pretty_reference }} go build -v
+```
+
+To run the application tests inside a container, run the following command:
```
Comment (dmpop) [inline src/bci_build/package/golang/README.md.j2:45]:
```suggestion
To run the application tests inside a container, use the following command:
```

File: src/bci_build/package/golang/README.md.j2 [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+The following additional tools are included in the image:
+
+- go{{ image.version }}-race
+- make
+- git-core
+
+{% if image.version.endswith('-openssl') %}
+## FIPS 140
+
+This image includes a FIPS 140 enabled Go toolchain using an OpenSSL.
```
Comment (dmpop) [inline src/bci_build/package/golang/README.md.j2:62]:
```suggestion
The image includes a FIPS 140 enabled Go toolchain using an OpenSSL.
```

File: src/bci_build/package/golang/README.md.j2 [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+You can add the following line to restricts all TLS configuration to FIPS-approved settings.
```
Comment (dmpop) [inline src/bci_build/package/golang/README.md.j2:64]: ### 
```suggestion
 To restrict all TLS configuration to FIPS-approved settings, add the following line:
```

File: src/bci_build/package/golang/README.md.j2 [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+WORKDIR /app
+
+COPY . .
+
+RUN go mod download && go mod verify
+RUN go build -v -o /usr/local/bin/app ./...
+
+CMD ["app"]
+```
```
Comment (dcermak) [inline src/bci_build/package/golang/README.md.j2:28]: We really shouldn't advertise the usage of the go container for runtime, it is enormous (> 500MB uncompressed!). We should be advertising either to deploy the app via scratch or at least bci-base, bci-minimal, bci-micro or bci-busybox. But an image build as described here will be giant, as this process leaves not only the go compiler, but also all the sources, dependencies & build results in the image.

File: src/bci_build/package/golang/README.md.j2 [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+$ podman build -t my-golang-app .
+$ podman run -it --rm my-golang-app
+```
+
+There may be occasions when it is inappropriate to run your application inside a container.
+
+To compile, but not run your application inside a container instance, run the following command:
+
+```ShellSession
+$ podman run --rm -v "$PWD":/app -w /app {{ image.pretty_reference }} go build -v
```
Comment (dcermak) [inline src/bci_build/package/golang/README.md.j2:42]:
```suggestion
$ podman run --rm -v "$PWD":/app:Z -w /app {{ image.pretty_reference }} go build -v
```

File: src/bci_build/package/golang/README.md.j2 [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+```
+
+To run the application tests inside a container, run the following command:
+
+```ShellSession
+$ podman run --rm -v "$PWD":/app -w /app {{ image.pretty_reference }} go test -v
```
Comment (dcermak) [inline src/bci_build/package/golang/README.md.j2:48]:
```suggestion
$ podman run --rm -v "$PWD":/app:Z -w /app {{ image.pretty_reference }} go test -v
```

File: src/bci_build/package/golang/README.md.j2 (hunk 7 above)
Comment (dcermak) [inline src/bci_build/package/golang/README.md.j2:62]:
```suggestion
This image includes a FIPS 140 enabled Go toolchain using OpenSSL.
```

**Review by dcermak — CHANGES_REQUESTED** [review]

File: src/bci_build/package/golang/README.md.j2 (hunk 9 above)
Comment (alexandrevicenzi) [reply src/bci_build/package/golang/README.md.j2:28]: Ok, that makes sense, not sure which image is better, or if it is better to not mention any images at all, and just say this image is for development only.

Comment (alexandrevicenzi) [reply src/bci_build/package/golang/README.md.j2]: I reworked this part, take a look.

File: src/bci_build/package/golang/README.md.j2 [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+
+WORKDIR /app
+
+COPY . .
+
+RUN go mod download && go mod verify
+RUN go build -v -o /usr/local/bin/app ./...
+
+CMD ["app"]
+```
```
[the remark this answers is not in the document]
Comment (alexandrevicenzi) [reply src/bci_build/package/golang/README.md.j2:24]: yes, given that we don't recommend using the same image, it does make sense to use multi-stage here.

**Review by dcermak — APPROVED** [review]
Thanks a lot!

**Pull request merged**